datasource/src/main/scala/quasar/blobstore/services.scala

```diff
@@ -17,17 +17,24 @@
 package quasar.blobstore
 
 import slamdata.Predef._
-import quasar.api.resource.{ResourceName, ResourcePath, ResourcePathType}
 
 import fs2.Stream
 
-trait Blobstore[F[_]] {
+object services {
```

Comment:
I think replacing these traits with type aliases of `Kleisli` (i.e. `type GetService[F, P] = Kleisli[F, P, Stream[F, Byte]]`) would allow you to get rid of all the implicit converters and `ops.service`. Instead you could define services with concrete types and leverage `<==<` and `>==>` to (contra)map input and output at the use site.